examples/jenkins/pipeline/maven-pipeline.yaml

```diff
@@ -0,0 +1,229 @@
+  name: maven-pipeline
+parameters:
+- name: APP_NAME
+  description: The name assigned to all of the application objects defined in this template.
+  displayName: Application Name
+  required: true
+  value: openshift-jee-sample
+- name: GIT_SOURCE_URL
+  description: The source URL for the application
+  displayName: Source URL
+  required: true
+  value: https://github.com/bparees/openshift-jee-sample.git
+- name: GIT_SOURCE_REF
+  description: The source Ref for the application
+  displayName: Source Ref
+  required: true
+  value: master
+- description: A secret string used to configure the GitHub webhook.
+  displayName: GitHub Webhook Secret
+  from: '[a-zA-Z0-9]{40}'
+  generate: expression
+  name: GITHUB_WEBHOOK_SECRET
+  required: true
+- description: A secret string used to configure the Generic webhook.
+  displayName: Generic Webhook Secret
+  from: '[a-zA-Z0-9]{40}'
+  generate: expression
+  name: GENERIC_WEBHOOK_SECRET
+  required: true
+objects:
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    labels:
+      app: ${APP_NAME}
+    name: ${APP_NAME}
+  spec: {}
+  status:
+    dockerImageRepository: ""
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    annotations:
+      pipeline.alpha.openshift.io/uses: '[{"name": "${NAME}", "namespace": "", "kind": "DeploymentConfig"}]'
+    creationTimestamp: null
+    labels:
+      name: ${APP_NAME}
+    name: ${APP_NAME}
+  spec:
+    strategy:
+      jenkinsPipelineStrategy:
+        jenkinsfile: |-
+          def appName="${APP_NAME}"
+          def project=""
+
+          node {
+            stage("Initialize") {
+              project = env.PROJECT_NAME
+            }
+          }
+
+          node("maven") {
+            stage("Checkout") {
+              git url: "${GIT_SOURCE_URL}", branch: "${GIT_SOURCE_REF}"
+            }
+            stage("Build WAR") {
+              sh "mvn clean package -Popenshift"
+              stash name:"war", includes:"target/ROOT.war"
+            }
+          }
+
+          node {
+            stage("Build Image") {
+              unstash name:"war"
+              sh "oc start-build ${appName}-docker --from-file=target/ROOT.war --follow -n ${project}"
+            }
+            stage("Deploy") {
+              openshiftDeploy deploymentConfig: appName, namespace: project
+            }
+          }
+      type: JenkinsPipeline
+    triggers:
+    - github:
+        secret: "${GITHUB_WEBHOOK_SECRET}"
+      type: GitHub
+    - generic:
+        secret: "${GENERIC_WEBHOOK_SECRET}"
+      type: Generic
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    labels:
+      app: ${APP_NAME}-docker
+    name: ${APP_NAME}-docker
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: ${APP_NAME}:latest
+    postCommit: {}
+    resources: {}
+    runPolicy: Serial
+    source:
+      dockerfile: |-
+        FROM wildfly
+        COPY ROOT.war /opt/app-root/src/target/ROOT.war
+        RUN $STI_SCRIPTS_PATH/assemble
```

Comment:
do you really need to run assemble vs just copying the war directly into the deployment dir?